ui/src/components/nodes/Code.tsx

```diff
@@ -278,19 +275,88 @@ export const ResultBlock = memo<any>(function ResultBlock({ id, layout }) {
                 fontSize: "0.8em",
-                flexDirection: "row",
-                alignItems: "center",
+                flexDirection: "column",
+                alignItems: "left",
                 borderTop: "1px solid rgb(214, 222, 230)",
               }}
             >
-              <Box component="pre" whiteSpace="pre-wrap">
-                {result.text}
-              </Box>
+              {results.map((res, i) => {
+                const combinedKey = `${res.type}-${i}`;
+                switch (res.type) {
+                  case "stream_stdout":
+                    return (
+                      <Box
+                        component="pre"
+                        whiteSpace="pre-wrap"
+                        key={combinedKey}
+                        sx={{ fontSize: "0.8em" }}
+                      >
+                        <Ansi>{res.text}</Ansi>
+                      </Box>
+                    );
+                  case "stream_stderr":
+                    return (
+                      <Box
+                        component="pre"
+                        whiteSpace="pre-wrap"
+                        key={combinedKey}
+                        sx={{ fontSize: "0.8em" }}
+                      >
+                        <Ansi>{res.text}</Ansi>
+                      </Box>
+                    );
+                  case "display_data":
+                    return (
+                      <Box
+                        component="pre"
+                        whiteSpace="pre-wrap"
+                        key={combinedKey}
+                      >
+                        {res.text}
+                        <img
+                          src={`data:image/png;base64,${res.image}`}
+                          alt="output"
+                        />
+                      </Box>
+                    );
+                  case "execute_result":
+                    return (
+                      <Box
+                        component="pre"
+                        whiteSpace="pre-wrap"
+                        key={combinedKey}
+                        sx={{
+                          fontSize: "0.8em",
+                          borderTop: "1px solid rgb(214, 222, 230)",
+                        }}
+                      >
+                        {res.text}
+                      </Box>
+                    );
+                  case "execute_reply":
+                    if (i == 0) {
+                      return (
+                        <Box
+                          component="pre"
+                          whiteSpace="pre-wrap"
+                          key={combinedKey}
+                          sx={{ fontSize: "0.8em" }}
+                        >
+                          {res.text}
+                        </Box>
+                      );
+                    } else {
+                      return <></>;
+                    }
+                  default:
+                    return <></>;
```

Comment:
The console warning is still there, caused by these two lines. Adding `key={combineKey}` will fix it.